File: backend/simple-api/src/main/java/com/example/simple_api/services/TaskService.java
package com.example.simple_api.services;


import com.example.simple_api.dto.TaskDTO;
import com.example.simple_api.repository.TaskRepository;
import org.springframework.stereotype.Service;
import com.example.simple_api.models.Task;

import java.util.List;

@Service
public class TaskService {

    private final TaskRepository taskRepository;

    private Task dtoToModel(TaskDTO dto) throws Exception {
        Task task = new Task();
        task.setDescription(dto.getDescription());
        task.setStatus(dto.getStatus());
        task.setTitle(dto.getTitle());
        return task;
    }

    public TaskService(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }

    public List<Task> getTasks() throws Exception {
        return taskRepository.getTasks();
    }

    public Task getTask(long id) throws Exception {
        return taskRepository.getTaskById(id);
    }

    public Task createTask(TaskDTO dto) throws Exception {
        return taskRepository.saveTask(dtoToModel(dto));
    }

    public Task updateTask(TaskDTO dto, long id) throws Exception {
        return taskRepository.updateTask(dtoToModel(dto), id);

    }

    public void deleteTaskById(long id) throws Exception {
        taskRepository.deleteTaskById(id);
    }

//    public Task updateTaskStatus(int id, int status) {
//
//    }


}

File: frontend/simple-api/src/services/TaskService.ts
import api from "@/services/httpClient"
import { Task } from "@/components/models/Task";

export class TaskService{

  async getTaskById(id: number): Promise<Task> {
    return (await api.get("/task/" + id)).data;
  }

  async getTasks(): Promise<Task[]> {
    return (await api.get("/tasks")).data;
  }

  async createTask(task: Task): Promise<Task> {
    return (await api.post("/task/save", task)).data;
  }

  async updateTask(task: Task, id: number): Promise<Task> {
    return (await api.put("/task/" + id, task)).data;
  }

  async deleteTask(id: number) {
    return (await api.delete("/task/" + id)).data;
  }

}
